Adding summary script for COVID runs

OlintoYale opened the pull request:
This script could be used for the  analysis summary for PatientSamples: Two files are generated: 1: Summary and 2: RunParameters.
Only needs an arguments: run name, ie.  Rscript RCovidSummary.R UT-A01290220822
Conditions/precautions:
      a: Analysis completed and Cecret/Lineage_report.csv completed before running this script
      b: The sample IDs, generated with a random number generator, will be different each time this script is run.

OlintoYale committed:
Initial commit

jwarnn commented on RCovidSummary.R:
```diff
@@ -0,0 +1,94 @@
+# Author: JLinares/2022
+# Usage: 1. analysis summary for PatientSamples: Two files are generated: Summary and RunParameters
+#        2. Arguments: run name
+#        3. Example: Rscript / . . . ./summary_v2.R UT-A01290-220416
```
Just a little thing; I had to look for a run that was still in the NGS. I used UT-A01290-220706.

erinyoung replied:
This line needs to be updated to 
```
Example: Rscript / . . . ./RCovidSummary.R UT-A01290-220416
```

jwarnn commented on RCovidSummary.R:
```diff
@@ -0,0 +1,94 @@
+#        3. Example: Rscript / . . . ./summary_v2.R UT-A01290-220416
+#        4. Connditions/precautions:
+#           a: Analysis completed and Cecret/Lineage_report.csv completed before running this script
+#           b: The samplesIds, generated using a Randon Number Generator, will be differents everytime this sctript is ran.
```
I would also add where the output is going to be. I know its at the bottom but I missed it when I ran the script and thought that it would be in the directory I ran the script in.

jwarnn reviewed (approved):
Script runs on Linux workstation, and the outputs look correct. I had a few comments on the usage section.

OlintoYale commented:
John,
Thanks for revising the script.

The output is going to be in the same folder where the run is.
i.e. If we use the run UT-A01290-220822, the output files will be at
/Volumes/NGS/Analysis/covidseq/*UT-A01290-220822*/Rsumcovidseq

Thanks,

Olinto

On Fri, Sep 2, 2022 at 10:28 AM John Arnn ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> Script run on Linux workstation, and the outputs look correct. I had a few
> comments on the usage section.
> ------------------------------
>
> In RCovidSummary.R
> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/16#discussion_r961832355>
> :
>
> > @@ -0,0 +1,94 @@
> +#!/usr/bin/env Rscript
> +# Author: JLinares/2022
> +# Usage: 1. analysis summary for PatientSamples: Two files are generated: Summary and RunParameters
> +#        2. Arguments: run name
> +#        3. Example: Rscript / . . . ./summary_v2.R UT-A01290-220416
>
> Just a little thing; I had to look for a run that was still in the NGS. I
> used UT-A01290-220706.
> ------------------------------
>
> In RCovidSummary.R
> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/16#discussion_r961833175>
> :
>
> > @@ -0,0 +1,94 @@
> +#!/usr/bin/env Rscript
> +# Author: JLinares/2022
> +# Usage: 1. analysis summary for PatientSamples: Two files are generated: Summary and RunParameters
> +#        2. Arguments: run name
> +#        3. Example: Rscript / . . . ./summary_v2.R UT-A01290-220416
> +#        4. Connditions/precautions:
> +#           a: Analysis completed and Cecret/Lineage_report.csv completed before running this script
> +#           b: The samplesIds, generated using a Randon Number Generator, will be differents everytime this sctript is ran.
>
> I would also add where the output is going to be. I know its at the bottom
> but I missed it when I ran the script and thought that it would be in the
> directory I ran the script in.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/16#pullrequestreview-1095092865>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUISDYNYYGURUTKDDMQX2HTV4ITK3ANCNFSM6AAAAAAQCYCVGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
*************** IMPORTANT MESSAGE ***************
This message, including 
any attachments, may contain confidential information intended for a 
specific individual and purpose, and is protected by law.  If you are not 
the intended recipient, delete this message, including from trash, and 
notify me by telephone or email.


If you are not the intended recipient, 
any distributions or copying of this message, or the taking of any action 
based on its content is strictly prohibited.

OlintoYale committed:
Usage for RCovidSummary.R script

OlintoYale committed:
Update RCovidSummary.R

1. Fixed usage on line 3: Rscript RCovidSummary.R UT-A01290-220901
2. Added files store location.

erinyoung commented on README.md:
```diff
@@ -19,3 +19,12 @@ MiSeq_runs_basespace.sh
 ## daily_SARS-CoV-2_metadata.sh
 
 daily_SARS-CoV-2_metadata.sh checks the date every 4 hours. If it is a new day, this script runs the Dripping Rock nextflow workflow to get the daily SARS-CoV-2 metadata file. On Sundays, it also takes the latest 20 runs and 10 random runs and creates a phylogenetic tree. This is a script that is run in a screen on the Production account.
+
+ 
+## RCovidSummary.R
+
+This script shold be run after the analytics phase and the pangolin-lineage are completed. The script summaryzes the run and generated two files: RunParameters and RCovidseq_summary, that will be stored at the runpath/Rsumcovidseq folder. 
+
+RunParameters: Is the collection of the run static parameters: Wetlab assay, instrument type, software, pangolin version, Read Type, Number of Samples, number of Lanes, Chemistry , index adapters and the date of the run.
+
+Rcovidseq_summary: Is the collecton of the dynamic variables: sample_IDs, lineage, scorpion_call, num_actg, num_n, pass/fail.
```
I was looking at your readme, and I was wondering why it was so different than mine. Then I realized that https://github.com/UPHL-BioNGS/General_LW_scripts/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5 hasn't been merged, yet. Ooops!

I don't actually understand how or when to run this script, but here's my best guess (please change where applicable)

"""
### RCovidSummary.R

RCovidSummary.R should be run once the [dragen-covid-pipeline.sh](https://github.com/UPHL-BioNGS/dragen_scripts) has finished on a SARS-CoV-2 covidseq run (runs on the dragen server), and pangolin has been run on the resultant fasta files. The script summarizes the run and generates two files in the $runpath/Rsumcovidseq directory: RunParameters-$date.csv (values that are the same for each sample) and RCovidseq_summary-$date.csv (values that are specific to each sample)

```
Rscript RCovidSummary.R UT-A01290-220901
```

"""

OlintoYale replied:
That is better explanation to know how/when the script should be run.

OlintoYale committed:
Update RCovidSummary.R

Typo in column names

OlintoYale committed:
Update RCovidSummary.R

Included "Number of reads"  and "Mean quality score"  in the Rcovised_summary-$date.csv file

OlintoYale committed:
Update RCovidSummary.R

updated the path to SampleSheet

erinyoung commented:
Sorry, it's been a few weeks. 

I tried running this: 
```
$ Rscript RCovidSummary.R UT-A01290-220930
/home/linuxbrew/.linuxbrew/Cellar/r/4.2.0/lib/R/bin/exec/R: error while loading shared libraries: libicuuc.so.70: cannot open shared object file: No such file or directory
```
and it looks like the R libraries weren't installed for prod


Also, could we update the readme with how this script is intended to be used?

OlintoYale commented:
I am revising/updating the script and the libraries today.

erinyoung committed:
Merge branch 'main' into RSummary

OlintoYale commented:
I closed the PR ( #16) on July 26-2023. It has been reopen on PR 25, long time ago, that include major modifications.
